tests/system_tests/test_plans.py

```diff
@@ -0,0 +1,86 @@
+import os
+
+import pytest
+from blueapi.client.client import BlueapiClient
+from blueapi.client.event_bus import AnyEvent
+from blueapi.core.bluesky_types import DataEvent
+from blueapi.worker.event import TaskStatus, WorkerEvent, WorkerState
+from blueapi.worker.task import Task
+
+_DISABLE_SIDE_EFFECTS = bool(os.environ.get("DISABLE_SIDE_EFFECTS", 0))
+_DISABLE_SIDE_EFFECTS_MESSAGE = """
+    This test would cause side effects on the beamline, it has been disabled
+    so as not to interfere with operation. To run tests that may interfere with
+    the beamline export DISABLE_SIDE_EFFECTS=0 or add to add to env in pyproject.toml
+    """
+disable_side_effects = pytest.mark.skipif(
+    _DISABLE_SIDE_EFFECTS, reason=_DISABLE_SIDE_EFFECTS_MESSAGE
+)
+
+
+def _check_all_events(all_events: list[AnyEvent]):
+    assert (
+        isinstance(all_events[0], WorkerEvent) and all_events[0].task_status is not None
+    )
+    task_id = all_events[0].task_status.task_id
+    # First event is WorkerEvent
+    assert all_events[0] == WorkerEvent(
+        state=WorkerState.RUNNING,
+        task_status=TaskStatus(
+            task_id=task_id,
+            task_complete=False,
+            task_failed=False,
+        ),
+    )
+
+    assert all(
+        isinstance(event, DataEvent) for event in all_events[1:-2]
+    ), "Middle elements must be DataEvents."
+
+    # Last 2 events are WorkerEvent
+    assert all_events[-2:] == [
+        WorkerEvent(
+            state=WorkerState.IDLE,
+            task_status=TaskStatus(
+                task_id=task_id,
+                task_complete=False,
+                task_failed=False,
+            ),
+        ),
+        WorkerEvent(
+            state=WorkerState.IDLE,
+            task_status=TaskStatus(
+                task_id=task_id,
+                task_complete=True,
+                task_failed=False,
+            ),
+        ),
+    ]
+
+
+@pytest.mark.parametrize("device", ["sample_stage", "panda", "det"])
+def test_has_minimal_devices(client: BlueapiClient, device: str):
```

Comment:
Nit: I think this would look better on the test display

```suggestion
def test_device_present(client: BlueapiClient, device: str):
```

```
test_device_present::panda
test_device_present::det
etc...
```